src/hotspot/share/compiler/compilerDirectives.cpp

```diff
@@ -735,6 +735,29 @@ void DirectivesStack::release(CompilerDirectives* dir) {
   }
 }
 
+bool DirectivesStack::hasMatchingDirectives(const methodHandle& method, bool top_only) {
+  assert(_depth > 0, "Must never be empty");
+  bool match_found = false;
+  {
+    MutexLocker locker(DirectivesStack_lock, Mutex::_no_safepoint_check_flag);
+
+    CompilerDirectives* dir = _top;
+    assert(dir != nullptr, "Must be initialized");
+
+    while (dir != nullptr) {
+      if (!dir->is_default_directive() && dir->match(method)) {
+        match_found = true;
+        break;
```

Comment:
`match_found` is redundant: for better readability, you may just return true. Curly braces around MutexLocker won't be needed either.

Reply:
Thanks, that's indeed simpler.